### PR TITLE
Add json payload flow model node

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Part of [Implementing data flow nodes](https://github.com/wso2-enterprise/eggplant-project/issues/116).

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java [hunk 1]
```diff
@@ -683,9 +685,9 @@ public PropertiesBuilder<T> onErrorVariable(TypedBindingPatternNode typedBinding
             }
             Property type = propertyBuilder
                     .metadata()
-                        .label(Property.ON_ERROR_TYPE_LABEL)
-                        .description(Property.ON_ERROR_TYPE_DOC)
-                        .stepOut()
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java:688]: This formatting is expected

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java [hunk 2]
```diff
@@ -736,17 +740,38 @@ public void visit(ListConstructorExpressionNode listCtrExprNode) {
     }
 
     private void handleConstructorExpressionNode(ExpressionNode constructorExprNode, String typeDescription) {
-        SyntaxKind kind = constructorExprNode.parent().kind();
+        NonTerminalNode parent = constructorExprNode.parent();
+        SyntaxKind kind = parent.kind();
         if (kind == SyntaxKind.LOCAL_VAR_DECL || kind == SyntaxKind.MODULE_VAR_DECL) {
+            Optional<Symbol> parentSymbol = semanticModel.symbol(parent);
+            if (parentSymbol.isPresent() && CommonUtils.getRawType(
+                    ((VariableSymbol) parentSymbol.get()).typeDescriptor()).typeKind() == TypeDescKind.JSON) {
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:748]: Can we extract this to a private static method, and reuse that method

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java [hunk 3]
```diff
@@ -745,20 +749,44 @@ public void visit(ListConstructorExpressionNode listCtrExprNode) {
...
-                    .variable(((AssignmentStatementNode) constructorExprNode.parent()).varRef());
+                    .variable(((AssignmentStatementNode) parent).varRef());
         }
     }
 
+    private static boolean isVarTypeJson(Symbol parentSymbol) {
+        return CommonUtils.getRawType(
+                ((VariableSymbol) parentSymbol).typeDescriptor()).typeKind() == TypeDescKind.JSON;
+    }
+
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:789]: Shall we move this to the `CommoUtils`?

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java [hunk 4]
```diff
@@ -745,20 +749,44 @@ public void visit(ListConstructorExpressionNode listCtrExprNode) {
...
         if (kind == SyntaxKind.LOCAL_VAR_DECL || kind == SyntaxKind.MODULE_VAR_DECL) {
+            Optional<Symbol> parentSymbol = semanticModel.symbol(parent);
+            if (parentSymbol.isPresent() && isVarTypeJson(parentSymbol.get())) {
+                startNode(FlowNode.Kind.JSON_PAYLOAD)
+                        .metadata()
+                        .description(JSONPayload.DESCRIPTION)
+                        .stepOut()
+                        .properties().expression(constructorExprNode);
+                return;
+            }
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:763]: Seems like this is common for both conditions. Better to extract this out and have this as the first if condition.

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java (hunk 3 above)
Comment (LakshanWeerasinghe) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java:789]: This function doesn't provide any generic functionality which is useful elsewhere. So I think better to keep it here.

**Review by nipunayf — APPROVED** [review]

**Comment by nipunayf** [comment]
Addressed remaining changes via: https://github.com/ballerina-platform/ballerina-dev-tools/pull/414

**Pull request merged**